mvn-extension/src/main/java/org/arquillian/smart/testing/mvn/ext/dependencies/StrategyDependencyResolver.java

```diff
@@ -67,26 +67,46 @@ private Properties loadDefaultMapping() {
         return properties;
     }
 
-    private Map<String, Dependency> transformToDependencies(Properties properties) {
+    private Map<String, Dependency> transformToDependencies(Properties properties, boolean excludeTrainsitive) {
         return properties
             .stringPropertyNames()
             .stream()
             .filter(key -> key.startsWith(SMART_TESTING_STRATEGY_PREFIX))
             .map(String::valueOf)
             .collect(Collectors.toMap(this::filterPrefix,
-                key -> {
-                    final String[] gav = ((String) properties.get(key)).split(":");
-                    final Dependency dependency = new Dependency();
-                    dependency.setGroupId(gav[0]);
-                    dependency.setArtifactId(gav[1]);
-                    String version = ExtensionVersion.version().toString();
-                    if (gav.length == 3) {
-                        version = gav[2];
-                    }
-                    dependency.setVersion(version);
-                    dependency.setScope("runtime");
-                    return dependency;
-                }));
+                key -> createDependency((String) properties.get(key), excludeTrainsitive)));
+    }
+
+    private Dependency createDependency(String coordinatesString, boolean excludeTrainsitive) {
```

Comment:
Maybe we can move it to external class called `MavenCoordinatesResolver` and also provide unit tests for just this feature?

Reply:
I had the same feeling when I was writing it. But I was too lazy :-D

Reply:
Ok, I moved it to separated class and also created a test with custom assertion for it